server/src/main/kotlin/suwayomi/tachidesk/manga/MangaAPI.kt

```diff
@@ -77,11 +77,13 @@ object MangaAPI {
             get("", CategoryController::categoryList)
             post("", CategoryController::categoryCreate)
 
+            // The order here is important {categoryId} needs to be applied last 
+            // or throws a NumberFormatException
+            patch("reorder", CategoryController::categoryReorder)
+
```

Comment:
So a Little down below there's a patch route, which starts with {categoryId}. Javelin is trying to apply the routes in order of definition. So when the patch with categoryId is defined before the reorder, then reorder will never be called because the wild card {categoryId} will always match before.